torch/csrc/autograd/profiler_utils.cpp

```diff
@@ -0,0 +1,91 @@
+#include <torch/csrc/autograd/profiler_utils.h>
+
+namespace torch { namespace autograd { namespace profiler {
+
+const size_t CONV2D_STRIDE = 3;
+const size_t CONV2D_PADDING = 4;
+const size_t CONV2D_DILATION = 5;
+const size_t CONV2D_GROUPS = 6;
+
+void saveExtraArgs(std::unordered_map<std::string, c10::IValue> &extra_args, const at::RecordFunction& fn) {
+    // switch the type of fn, if convolution or gemm, fill-in extra_args_, then return
+    std::vector<c10::IValue> inputs = fn.inputs();
+    if(fn.name() == at::StringView("aten::conv2d")) {
+      at::Tensor input = inputs[0].toTensor();
```

Comment:
if you get failures in toTensor then it might mean that the actual inputs are not the ones you expect here

Reply:
you need to make sure that we don't try to analyze inputs in cases when profiler runs without record_shapes=True